pkg/transaction/transaction.go

```diff
@@ -584,3 +588,17 @@ func (t *transactionService) TransactionFee(ctx context.Context, txHash common.H
 	}
 	return trx.Cost(), nil
 }
+
+func (t *transactionService) UnwrapRevertReason(ctx context.Context, req *TxRequest, err error) error {
+	if err == nil {
+		return nil
+	}
+
+	if res, cErr := t.Call(ctx, req); cErr == nil {
+		if reason, uErr := abi.UnpackRevert(res); uErr == nil {
```

Comment:
UnapckRevert will unpack abi encoded "Error(string)", but I do not think that it will unpack custom errors that are for example used in the redistribution contract. To unpack custom errors, contract abi is required, get the error with abi.ABI.ErrorByID using the first 4 bytes of the data and use the retuned *abi.Error to Unpack the error.

I did not try it, in this PR, but used this approach to get custom errors from the call traces.

Reply:
This function may benefit from unit tests with different types of revert errors to ensure that they are not silently ignored.

Reply:
Yes, this is for revert reason as specified in: https://solidity.readthedocs.io/en/latest/control-structures.html#revert

Reply:
Exactly, this is the one type of revert, there may be custom errors packed, and custom errors are used in our contracts. They supposed to be packed differently as they have arguments and custom name.

Reply:
Can the custom errors be rewritten into this generic form so the error extraction is unified?

Reply:
They can be just strings, but sometimes additional values are useful to be passed. In any case, solidity allows custom revert errors and I do not think that they are unpacked here.

Reply:
Ok, merging this. The custom errors need to be reviewed and handled in different PR.

Reply:
> Can the custom errors be rewritten into this generic form so the error extraction is unified?

actually no, this generic form wasted a lot of space so was turned into a less wasteful form of generating errors, which in turn saves gas and in the end money for users. Its not a big deal to use them just needs to be unpacked properly will show examples of what is the difference, should be easy enough.